### PR TITLE
strap lansible run with ansible playbook

### PR DESCRIPTION
**Pull request by songtingxu-okta**

### CODE REVIEW

File: lib/ansible.sh [hunk 1]
```diff
@@ -278,4 +278,60 @@ EOF
   )
 }
 
-set +a
\ No newline at end of file
+strap::ansible::playbook::run() {
+
+  local playbook_dir="$(pwd)/.strap/ansible/playbooks/default"
```
Comment (les-okta) [inline lib/ansible.sh]: You don't need to execute `pwd`.  Strap has `STRAP_WORKING_DIR` that you should reference directly (faster, lower overhead).

File: lib/ansible.sh [hunk 2]
```diff
@@ -278,4 +278,60 @@ EOF
...
+  local playbook_file= requirements_file= 
+
+  while (( "$#" )); do
+    [[ "$1" == --*=* ]] && set -- "${1%%=*}" "${1#*=}" "${@:2}" # normalize `--foo=bar` into `--foo bar`
+    case "$1" in
+      --playbook-dir|--with-playbook-dir)
+        playbook_dir="${2:-$(pwd)/.strap/ansible/playbooks/default}"
```
Comment (les-okta) [inline lib/ansible.sh]: again, no need for `pwd`

File: lib/ansible.sh (hunk 1 above)
Comment (songtingxu-okta) [reply lib/ansible.sh]: Good to know! Changed

File: lib/ansible.sh (hunk 2 above)
Comment (songtingxu-okta) [reply lib/ansible.sh]: Changed

File: lib/ansible.sh [hunk 3]
```diff
@@ -278,4 +278,60 @@ EOF
...
+  fi
+
+  playbook_file="${playbook_dir}/main.yml"
+  requirements_file="${playbook_dir}/meta/requirements.yml"
+  [[ -f "${requirements_file}" ]] || requirements_file="${playbook_dir}/requirements.yml"
+
+  (
+    cd "${playbook_dir}"
+    export ANSIBLE_LOG_PATH="${STRAP_ANSIBLE_LOG_FILE}"
+    [[ -f "${requirements_file}" ]] && strap::ansible-galaxy-install -r "${requirements_file}" --force
```
Comment (songtingxu-okta) [inline lib/ansible.sh]: `--force` to ensure we have the right version of roles installed, which is important when same role of other versions have been installed already.

**Pull request merged**